components/ReadFileDropContainer.js

```diff
@@ -12,8 +12,53 @@ const spec = {
   }
 }
 
-const collect = connect => ({ connectDropTarget: connect.dropTarget() })
+const collect = (connect, monitor) => ({ connectDropTarget: connect.dropTarget(), isOver: monitor.isOver() })
 
-const ReadFileDropContainer = props => props.connectDropTarget(props.children)
+class ReadFileDropContainer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { isFileOver: false }
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
You should not need to set this state right? Once the props are updated it should rerender...